source/common/router/router.cc

```diff
@@ -2096,6 +2102,38 @@ bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
   return false;
 }
 
+void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or_trailers,
```

Comment:
High level question: it is not clear to me whether this method only processes the load report Envoy gets from the upstream or is it also coupled with LRS usage?

From the name (and my expectation) it should work no matter if LRS is defined. But from the comments in the function it seems to be requiring LRS.
I think this needs to be clarified, as LRS should be decoupled from ORCA.

Reply:
Currently `maybeProcessOrcaLoadReport` will process ORCA load reports only for LRS reporting. 

Having said that LRS reporting is decoupled from ORCA in a sense, that LRS could report LoadMetricStats collected using different methods.

In a followup PR `maybeProcessOrcaLoadReport` will also invoke a callback if it is configured by LB Policy via LoadBalancerContext.

Reply:
IIUC one can have ORCA without configuring LRS.
Looking at the following code below:
```
const bool need_to_send_load_report =
      (host != nullptr) && (cluster_ != nullptr) && cluster_->lrsReportMetricNames().has_value();
  if (!need_to_send_load_report) {
    return;
  }
  ```
I gather that parsing the ORCA contents from the upstream is only done if a load-report is configured.

In the context of this PR I think this is ok, as long as this will be changed in the future to allow parsing the ORCA contents even if LRS isn't configured.

For now, I think it would suffice to add a comment to the function definition, explaining what does "processing" imply.